### PR TITLE
Zs addviewcheckpage #14782

### LINKED ISSUE #14782: Add view page for Check
This issue will require that we create separate routes for check statuses/notifications/endpoints, which render a version of AlertingHistoryPage that doesn't allow user to delete the `check=checkID` search parameter in the search box. So that user can not make choices that make the visualization irrelevant. 

Adding status indicators to the visualization is not part of this issue, but part of https://github.com/influxdata/influxdb/issues/14618

### PR DESCRIPTION
**Pull request by zoesteinkamp**
Closes #14782 

Describe your proposed changes here.

- [ ] [CHANGELOG.md](https://github.com/influxdata/influxdb/blob/master/CHANGELOG.md) updated with a link to the PR (not the Issue)
- [ ] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [ ] Rebased/mergeable
- [ ] Tests pass
- [ ] http/swagger.yml updated (if modified Go structs or API)
- [ ] Documentation updated or issue created (provide link to issue/pr)
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: ui/src/alerting/components/CheckHistory.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
+// Libraries
+import React, {useMemo, FC} from 'react'
+import {Page} from '@influxdata/clockface'
+import {connect} from 'react-redux'
+
+// Components
+import {ResourceIDsContext} from 'src/alerting/components/AlertHistoryIndex'
+import EventViewer from 'src/eventViewer/components/EventViewer'
+// import CheckHistoryControls from 'src/alerting/components/CheckHistoryControls'
```
Comment (ebb-tide) [inline ui/src/alerting/components/CheckHistory.tsx]: missed comment!

File: ui/src/alerting/components/CheckHistory.tsx [hunk 2]
```diff
@@ -0,0 +1,113 @@
+// Libraries
+import React, {useMemo, FC} from 'react'
+import {Page} from '@influxdata/clockface'
```
Comment (ebb-tide) [inline ui/src/alerting/components/CheckHistory.tsx]: this is a component not a library :)

File: ui/src/alerting/components/CheckHistoryVisualization.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+              table={get(giraffeResult, 'table')}
+              fluxGroupKeyUnion={get(giraffeResult, 'fluxGroupKeyUnion')}
+              loading={loading}
+              timeZone={timeZone}
+              viewProperties={view.properties}
+              statuses={statuses}
+            >
+              {config => <Plot config={config} />}
+            </CheckPlot>
+            {/* {<CheckHistoryStatuses .../>} */}
```
Comment (ebb-tide) [inline ui/src/alerting/components/CheckHistoryVisualization.tsx]: stray comment

File: ui/src/alerting/components/CheckHistoryVisualization.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+export const ResourceIDsContext = createContext<ResourceIDs>(null)
+
+interface OwnProps {
+  check: Check
+  timeZone: TimeZone
+}
+
+type Props = OwnProps
+//TODO maybe update submitToken when we know how
```
Comment (ebb-tide) [inline ui/src/alerting/components/CheckHistoryVisualization.tsx]: stray comment

File: ui/src/alerting/components/CheckHistoryControls.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+interface Props {
+  eventViewerProps: EventViewerChildProps
+}
+
+const CheckHistoryControls: FC<Props> = ({eventViewerProps}) => {
+  return (
+    <div className="alert-history-controls">
+      <div className="alert-history-controls--right">
+        {/* <BackToTopButton {...eventViewerProps} /> */}
```
Comment (ebb-tide) [inline ui/src/alerting/components/CheckHistoryControls.tsx]: stray comment

File: ui/src/index.tsx [hunk 6]
```diff
@@ -445,6 +446,10 @@ class Root extends PureComponent {
                             path="alert-history"
                             component={AlertHistoryIndex}
                           />
+                          <Route
```
Comment (ebb-tide) [inline ui/src/index.tsx]: ❤️

File: ui/src/alerting/components/CheckHistoryVisualization.tsx [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+const CheckHistoryVisualization: FC<Props> = ({check, timeZone}) => {
+  const view = createView<CheckViewProperties>(get(check, 'type', 'threshold'))
```
Comment (ebb-tide) [inline ui/src/alerting/components/CheckHistoryVisualization.tsx]: I think we ought to be able to assume check has type here.

File: ui/src/alerting/components/CheckHistoryControls.tsx [hunk 8]
```diff
@@ -0,0 +1,33 @@
+// Libraries
+import React, {FC} from 'react'
+
+// Components
+// import BackToTopButton from 'src/eventViewer/components/BackToTopButton'
```
Comment (ebb-tide) [inline ui/src/alerting/components/CheckHistoryControls.tsx]: stray comment- what do you think about making it a habit to look over your PR as if you were reviewing someone else's right before submitting? :)

File: ui/src/alerting/components/CheckHistory.tsx [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+                <div className="alert-history-page--header">
+                  <Page.Title
+                    title="Check Statuses"
+                    testID="alert-history-title"
+                  />
+                  <AlertHistoryQueryParams
+                    searchInput={props.state.searchInput}
+                    historyType={historyType}
+                  />
+                  {/* <CheckHistoryControls eventViewerProps={props} /> */}
```
Comment (ebb-tide) [inline ui/src/alerting/components/CheckHistory.tsx]: stray comment

File: ui/src/alerting/components/CheckHistoryVisualization.tsx [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+
+// Types
+import {Check, TimeZone, CheckViewProperties} from 'src/types'
+import TimeSeries from 'src/shared/components/TimeSeries'
+import {createView} from 'src/shared/utils/view'
+import {checkResultsLength} from 'src/shared/utils/vis'
+import {getTimeRangeVars} from 'src/variables/utils/getTimeRangeVars'
+import {TimeRange} from 'src/types'
+
+interface ResourceIDs {
```
Comment (ebb-tide) [inline ui/src/alerting/components/CheckHistoryVisualization.tsx]: we are defining this interface in two places- would you mind defining it in one place and then importing it in the other?

File: ui/src/alerting/components/CheckHistory.tsx [hunk 11]
```diff
@@ -0,0 +1,113 @@
+// Libraries
+import React, {useMemo, FC} from 'react'
+import {Page} from '@influxdata/clockface'
+import {connect} from 'react-redux'
+
+// Components
+import {ResourceIDsContext} from 'src/alerting/components/AlertHistoryIndex'
```
Comment (ebb-tide) [inline ui/src/alerting/components/CheckHistory.tsx]: this isn't a component, lets put it in another section plz

File: ui/src/alerting/components/CheckHistory.tsx (hunk 1 above)
Comment (ebb-tide) [reply ui/src/alerting/components/CheckHistory.tsx]: wait- why isn't this being imported? this is the search bar I think it should be in the page? If you don't intend to have the search bar showing let's not create the page as a part of this PR.

Comment (zoesteinkamp) [reply ui/src/alerting/components/CheckHistory.tsx]: I reinstated the search bar, i think it was commented out from when we discussed if it worked

File: ui/src/alerting/components/CheckHistory.tsx (hunk 11 above)
Comment (zoesteinkamp) [reply ui/src/alerting/components/CheckHistory.tsx]: Found it imported in other files as context, so added that section

File: ui/src/alerting/components/CheckHistoryVisualization.tsx (hunk 7 above)
Comment (zoesteinkamp) [reply ui/src/alerting/components/CheckHistoryVisualization.tsx]: Removed!

File: ui/src/alerting/components/CheckHistoryVisualization.tsx (hunk 10 above)
Comment (zoesteinkamp) [reply ui/src/alerting/components/CheckHistoryVisualization.tsx]: Im confused by this, are you referring to the time range, or the interface resource IDs?

**Pull request merged**